docs/container-contract.md

```diff
@@ -0,0 +1,48 @@
+# Container Contract
+
+Substratus orchestrates machine learning operations without requiring any language-specific library dependencies. As long as a container satisfies the "contract", that container can be orchestrated with Substratus. This contract is designed to impose as few requirements as possible.
+
+## Workdir
+
+The workdir MUST be set to `/content`.
+
+```Dockerfile
+WORKDIR /content
+```
+
+### Jupyter
+
+This requirement applies to Model, Dataset, and Notebook containers.
+
+The `notebook.sh` script MUST be located in `$PATH`. It is recommended that this script is stored in `/content/scripts`.
+
+* Starts a Jupyter Lab/Notebook environment.
+* Serve on port `8888`.
+* Respects the `NOTEBOOK_TOKEN` environment variable.
+
+### Directory Structure
+
+```
+/content/         # Working directory.
+  data/           # Location where Datasets will be mounted (reading and loading).
+  src/            # Source code (*.py, *.ipynb) for loading, training, etc.
+  logs/           # Output of building/training jobs for debugging purposes.
+  model/          # Location to store the resulting model from loading or training.
+  saved-model/    # Location where a previously saved model will be mounted.
+```
+
+### Parameters
+
+Substratus provides params as a file (`/content/params.json`) and as environment variables to containers.
+
+For example, the below Substratus Model spec will pass `abc: xyz`
+to the container using the environment variable `PARAM_ABC=xyz`:
```

Comment:
what's the reason for not converting it to PARAMS? this confused me and didn't feel natural that the letter S gets removed.

Reply:
NOTE: Terraform allows for specifying `*.tfvars` files which then translate into `TF_VAR_{name}` environment variables, kind of like our use case: https://developer.hashicorp.com/terraform/cli/config/environment-variables#tf_var_name

Reply:
I think `PARAM` is fine. My only q: why have `PARAM_` at all instead of `XYZ=abc`?

Reply:
B/c lowercase env vars are weird